packages/databricks-vscode/src/configuration/AzureCliCheck.ts

```diff
@@ -23,13 +26,32 @@ export interface ErrorResult {
     error: Error;
 }
 
+class OrchestrationLoopError extends Error {
+    constructor() {
+        super("OrchestrationLoopError");
+    }
+}
+
 async function orchestrate<S, KEYS extends string>(
     steps: Record<KEYS, Step<S, KEYS>>,
-    start: KEYS
+    start: KEYS,
+    maxSteps = 20,
+    logger?: NamedLogger
 ): Promise<S> {
+    let counter = 0;
+
     let step: KEYS | undefined = start;
     while (step && steps[step]) {
+        counter += 1;
+        if (counter > maxSteps) {
+            throw new OrchestrationLoopError();
+        }
         const result: StepResult<S, KEYS> = await steps[step]();
+        logger?.info(
+            `Azire CLI check: ${step}`,
+            util.inspect(result, {depth: 5})
```

Comment:
Logger will automatically do inspect internally. So this should be enough
```suggestion
            result
```

Reply:
We should infact do a logger?.error for errors because that handles non enumerable fields in errors as well.